internal/attestations/attestations.go

```diff
@@ -0,0 +1,161 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package attestations
+
+import (
+	"errors"
+
+	"github.com/gittuf/gittuf/internal/gitinterface"
+	"github.com/gittuf/gittuf/internal/rsl"
+	"github.com/gittuf/gittuf/internal/third_party/go-git"
+	"github.com/gittuf/gittuf/internal/third_party/go-git/plumbing"
+	"github.com/gittuf/gittuf/internal/third_party/go-git/plumbing/filemode"
+	"github.com/gittuf/gittuf/internal/third_party/go-git/plumbing/object"
+)
+
+const (
+	Ref                         = "refs/gittuf/attestations"
+	authorizationsTreeEntryName = "authorizations"
+	initialCommitMessage        = "Initial commit"
+	defaultCommitMessage        = "Update attestations"
+)
+
+var ErrAttestationsExist = errors.New("cannot initialize attestations namespace as it exists already")
+
+// InitializeNamespace creates a namespace to store attestations for
+// verification with gittuf. The ref is created with an initial, unsigned commit
+// that is unsigned.
+func InitializeNamespace(repo *git.Repository) error {
+	if ref, err := repo.Reference(plumbing.ReferenceName(Ref), true); err != nil {
+		if !errors.Is(err, plumbing.ErrReferenceNotFound) {
+			return err
+		}
+	} else if !ref.Hash().IsZero() {
+		return ErrAttestationsExist
+	}
+
+	treeHash, err := gitinterface.WriteTree(repo, nil)
+	if err != nil {
+		return err
+	}
+
+	_, err = gitinterface.Commit(repo, treeHash, Ref, initialCommitMessage, false)
+	return err
+}
+
+// Attestations tracks all the attestations in a gittuf repository.
+type Attestations struct {
+	authorizations map[string]plumbing.Hash // refPath/from-to -> blob ID
+}
+
+// LoadCurrentAttestations inspects the repository's attestations namespace and
+// loads the current attestations.
+func LoadCurrentAttestations(repo *git.Repository) (*Attestations, error) {
+	entry, _, err := rsl.GetLatestReferenceEntryForRef(repo, Ref)
+	if err != nil {
+		if !errors.Is(err, rsl.ErrRSLEntryNotFound) {
+			return nil, err
+		}
+
+		return &Attestations{}, nil
+	}
+
+	return LoadAttestationsForEntry(repo, entry)
+}
+
+// LoadAttestationsForEntry loads the repository's attestations for a particular
+// RSL entry for the attestations namespace.
+func LoadAttestationsForEntry(repo *git.Repository, entry *rsl.ReferenceEntry) (*Attestations, error) {
+	if entry.RefName != Ref {
+		return nil, rsl.ErrRSLEntryDoesNotMatchRef
+	}
+
+	attestationsCommit, err := gitinterface.GetCommit(repo, entry.TargetID)
+	if err != nil {
+		return nil, err
+	}
+
+	attestationsRootTree, err := gitinterface.GetTree(repo, attestationsCommit.TreeHash)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(attestationsRootTree.Entries) == 0 {
+		// This happens in the initial commit for the attestations namespace,
+		// with no entries in the tree at all
+		// An RSL entry isn't created by default for that entry but if one is,
+		// we don't want to panic.
```

Comment:
Do you mean the `Commit()` and other methods on `Attestations`?

Reply:
I think my original thought was that this was an unexpected case, but re-reading it seems like this might just be the initial state?

maybe rephrase this as:

```go
	if len(attestationsRootTree.Entries) == 0 {
		// This happens in the initial commit for the attestations namespace,
		// where there are no entries in the tree yet.
		// This is expected, and there is nothing more to check so return a zero Attestations state.
		return &Attestations{}, nil
	}
```

(added as suggestion)